StyleClassedTextArea + PageUp/PageDown -> NoSuchElementException + IndexOutOfBoundsException

I have a StyleClassedTextArea:

        textArea = new StyleClassedTextArea();
        textArea.getStyleClass().add("log-text-area");
        scrollPane = new VirtualizedScrollPane(textArea);
	textArea.wrapTextProperty().bindBidirectional(lineWrapButton.selectedProperty());
        textArea.setContextMenu(textAreaMenu);
        
        textArea.setEditable(false);
        textArea.setParagraphGraphicFactory(LineNumberFactory.get(textArea));
        textArea.setPadding(new Insets(0, 0, 0, 0));
        VBox.setVgrow(scrollPane, Priority.ALWAYS);

When text area has about 3000 lines I press and hold PageUp/PageDown I get a lot of NoSuchElementException  and IndexOutOfBoundsException exceptions. Is this a bug or I do something wrong?

```
2023-11-12 15:12:40.979 [ERROR] [JavaFX Application Thread] com.foo.FxPlatform - Error in FxApplicationjava.util.NoSuchElementException
	at reactfx@2.0-M5/org.reactfx.util.SparseList$AbsentSegment.getOrThrow(SparseList.java:78)
	at reactfx@2.0-M5/org.reactfx.util.FingerTree.lambda$get$2(FingerTree.java:1027)
	at reactfx@2.0-M5/org.reactfx.util.BiIndex.map(BiIndex.java:15)
	at reactfx@2.0-M5/org.reactfx.util.FingerTree.get(FingerTree.java:1027)
	at reactfx@2.0-M5/org.reactfx.util.SparseList.getPresent(SparseList.java:259)
	at reactfx@2.0-M5/org.reactfx.collection.MemoizationListImpl$MemoizedView.get(MemoizationList.java:41)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledArea.visibleParToAllParIndex(GenericStyledArea.java:990)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.ViewActions.lastVisibleParToAllParIndex(ViewActions.java:289)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledArea.nextPage(GenericStyledArea.java:1422)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledAreaBehavior.lambda$static$10(GenericStyledAreaBehavior.java:107)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate.lambda$consume$0(InputMapTemplate.java:221)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.PatternActionTemplate.lambda$null$1(InputMapTemplate.java:425)
	at java.base/java.util.Optional.map(Optional.java:260)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.PatternActionTemplate.lambda$getInputHandlerTemplateMap$2(InputMapTemplate.java:425)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:25)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:25)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate$1.lambda$null$0(InputMapTemplate.java:202)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate$HandlerTemplateConsumer$1.lambda$accept$0(InputMapTemplate.java:103)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.InputHandler.handle(InputHandler.java:50)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:247)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
	at javafx.base@19-ea/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
	at javafx.base@19-ea/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base@19-ea/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.base@19-ea/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.graphics@19-ea/javafx.scene.Scene$KeyHandler.process(Scene.java:4089)
	at javafx.graphics@19-ea/javafx.scene.Scene.processKeyEvent(Scene.java:2146)
	at javafx.graphics@19-ea/javafx.scene.Scene$ScenePeerListener.keyEvent(Scene.java:2614)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:218)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:150)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleKeyEvent$1(GlassViewEventHandler.java:250)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:424)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleKeyEvent(GlassViewEventHandler.java:249)
	at javafx.graphics@19-ea/com.sun.glass.ui.View.handleKeyEvent(View.java:542)
	at javafx.graphics@19-ea/com.sun.glass.ui.View.notifyKey(View.java:966)
	at javafx.graphics@19-ea/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at javafx.graphics@19-ea/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:316)
	at java.base/java.lang.Thread.run(Thread.java:1583)

2023-11-12 15:12:41.172 [ERROR] [JavaFX Application Thread] com.foo.FxPlatform - Error in FxApplicationjava.util.NoSuchElementException
	at reactfx@2.0-M5/org.reactfx.util.SparseList$AbsentSegment.getOrThrow(SparseList.java:78)
	at reactfx@2.0-M5/org.reactfx.util.FingerTree.lambda$get$2(FingerTree.java:1027)
	at reactfx@2.0-M5/org.reactfx.util.BiIndex.map(BiIndex.java:15)
	at reactfx@2.0-M5/org.reactfx.util.FingerTree.get(FingerTree.java:1027)
	at reactfx@2.0-M5/org.reactfx.util.SparseList.getPresent(SparseList.java:259)
	at reactfx@2.0-M5/org.reactfx.collection.MemoizationListImpl$MemoizedView.get(MemoizationList.java:41)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledArea.visibleParToAllParIndex(GenericStyledArea.java:990)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.ViewActions.lastVisibleParToAllParIndex(ViewActions.java:289)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledArea.nextPage(GenericStyledArea.java:1422)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledAreaBehavior.lambda$static$10(GenericStyledAreaBehavior.java:107)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate.lambda$consume$0(InputMapTemplate.java:221)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.PatternActionTemplate.lambda$null$1(InputMapTemplate.java:425)
	at java.base/java.util.Optional.map(Optional.java:260)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.PatternActionTemplate.lambda$getInputHandlerTemplateMap$2(InputMapTemplate.java:425)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:25)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:25)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate$1.lambda$null$0(InputMapTemplate.java:202)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate$HandlerTemplateConsumer$1.lambda$accept$0(InputMapTemplate.java:103)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.InputHandler.handle(InputHandler.java:50)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:247)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
	at javafx.base@19-ea/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
	at javafx.base@19-ea/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base@19-ea/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.base@19-ea/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.graphics@19-ea/javafx.scene.Scene$KeyHandler.process(Scene.java:4089)
	at javafx.graphics@19-ea/javafx.scene.Scene.processKeyEvent(Scene.java:2146)
	at javafx.graphics@19-ea/javafx.scene.Scene$ScenePeerListener.keyEvent(Scene.java:2614)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:218)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:150)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleKeyEvent$1(GlassViewEventHandler.java:250)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:424)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleKeyEvent(GlassViewEventHandler.java:249)
	at javafx.graphics@19-ea/com.sun.glass.ui.View.handleKeyEvent(View.java:542)
	at javafx.graphics@19-ea/com.sun.glass.ui.View.notifyKey(View.java:966)
	at javafx.graphics@19-ea/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at javafx.graphics@19-ea/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:316)
	at java.base/java.lang.Thread.run(Thread.java:1583)

2023-11-12 15:12:41.366 [ERROR] [JavaFX Application Thread] com.foo.FxPlatform - Error in FxApplicationjava.util.NoSuchElementException
	at reactfx@2.0-M5/org.reactfx.util.SparseList$AbsentSegment.getOrThrow(SparseList.java:78)
	at reactfx@2.0-M5/org.reactfx.util.FingerTree.lambda$get$2(FingerTree.java:1027)
	at reactfx@2.0-M5/org.reactfx.util.BiIndex.map(BiIndex.java:15)
	at reactfx@2.0-M5/org.reactfx.util.FingerTree.get(FingerTree.java:1027)
	at reactfx@2.0-M5/org.reactfx.util.SparseList.getPresent(SparseList.java:259)
	at reactfx@2.0-M5/org.reactfx.collection.MemoizationListImpl$MemoizedView.get(MemoizationList.java:41)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledArea.visibleParToAllParIndex(GenericStyledArea.java:990)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.ViewActions.lastVisibleParToAllParIndex(ViewActions.java:289)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledArea.nextPage(GenericStyledArea.java:1422)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledAreaBehavior.lambda$static$10(GenericStyledAreaBehavior.java:107)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate.lambda$consume$0(InputMapTemplate.java:221)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.PatternActionTemplate.lambda$null$1(InputMapTemplate.java:425)
	at java.base/java.util.Optional.map(Optional.java:260)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.PatternActionTemplate.lambda$getInputHandlerTemplateMap$2(InputMapTemplate.java:425)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:25)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:25)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate$1.lambda$null$0(InputMapTemplate.java:202)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate$HandlerTemplateConsumer$1.lambda$accept$0(InputMapTemplate.java:103)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.InputHandler.handle(InputHandler.java:50)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:247)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
	at javafx.base@19-ea/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
	at javafx.base@19-ea/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base@19-ea/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.base@19-ea/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.graphics@19-ea/javafx.scene.Scene$KeyHandler.process(Scene.java:4089)
	at javafx.graphics@19-ea/javafx.scene.Scene.processKeyEvent(Scene.java:2146)
	at javafx.graphics@19-ea/javafx.scene.Scene$ScenePeerListener.keyEvent(Scene.java:2614)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:218)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:150)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleKeyEvent$1(GlassViewEventHandler.java:250)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:424)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleKeyEvent(GlassViewEventHandler.java:249)
	at javafx.graphics@19-ea/com.sun.glass.ui.View.handleKeyEvent(View.java:542)
	at javafx.graphics@19-ea/com.sun.glass.ui.View.notifyKey(View.java:966)
	at javafx.graphics@19-ea/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at javafx.graphics@19-ea/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:316)
	at java.base/java.lang.Thread.run(Thread.java:1583)

2023-11-12 15:12:41.398 [ERROR] [JavaFX Application Thread] com.foo.FxPlatform - Error in FxApplicationjava.util.NoSuchElementException
	at reactfx@2.0-M5/org.reactfx.util.SparseList$AbsentSegment.getOrThrow(SparseList.java:78)
	at reactfx@2.0-M5/org.reactfx.util.FingerTree.lambda$get$2(FingerTree.java:1027)
	at reactfx@2.0-M5/org.reactfx.util.BiIndex.map(BiIndex.java:15)
	at reactfx@2.0-M5/org.reactfx.util.FingerTree.get(FingerTree.java:1027)
	at reactfx@2.0-M5/org.reactfx.util.SparseList.getPresent(SparseList.java:259)
	at reactfx@2.0-M5/org.reactfx.collection.MemoizationListImpl$MemoizedView.get(MemoizationList.java:41)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledArea.visibleParToAllParIndex(GenericStyledArea.java:990)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.ViewActions.lastVisibleParToAllParIndex(ViewActions.java:289)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledArea.nextPage(GenericStyledArea.java:1422)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledAreaBehavior.lambda$static$10(GenericStyledAreaBehavior.java:107)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate.lambda$consume$0(InputMapTemplate.java:221)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.PatternActionTemplate.lambda$null$1(InputMapTemplate.java:425)
	at java.base/java.util.Optional.map(Optional.java:260)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.PatternActionTemplate.lambda$getInputHandlerTemplateMap$2(InputMapTemplate.java:425)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:25)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:25)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate$1.lambda$null$0(InputMapTemplate.java:202)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate$HandlerTemplateConsumer$1.lambda$accept$0(InputMapTemplate.java:103)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.InputHandler.handle(InputHandler.java:50)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:247)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
	at javafx.base@19-ea/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
	at javafx.base@19-ea/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base@19-ea/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.base@19-ea/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.graphics@19-ea/javafx.scene.Scene$KeyHandler.process(Scene.java:4089)
	at javafx.graphics@19-ea/javafx.scene.Scene.processKeyEvent(Scene.java:2146)
	at javafx.graphics@19-ea/javafx.scene.Scene$ScenePeerListener.keyEvent(Scene.java:2614)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:218)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:150)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleKeyEvent$1(GlassViewEventHandler.java:250)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:424)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleKeyEvent(GlassViewEventHandler.java:249)
	at javafx.graphics@19-ea/com.sun.glass.ui.View.handleKeyEvent(View.java:542)
	at javafx.graphics@19-ea/com.sun.glass.ui.View.notifyKey(View.java:966)
	at javafx.graphics@19-ea/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at javafx.graphics@19-ea/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:316)
	at java.base/java.lang.Thread.run(Thread.java:1583)

2023-11-12 15:12:41.797 [ERROR] [JavaFX Application Thread] com.foo.FxPlatform - Error in FxApplicationjava.util.NoSuchElementException
	at reactfx@2.0-M5/org.reactfx.util.SparseList$AbsentSegment.getOrThrow(SparseList.java:78)
	at reactfx@2.0-M5/org.reactfx.util.FingerTree.lambda$get$2(FingerTree.java:1027)
	at reactfx@2.0-M5/org.reactfx.util.BiIndex.map(BiIndex.java:15)
	at reactfx@2.0-M5/org.reactfx.util.FingerTree.get(FingerTree.java:1027)
	at reactfx@2.0-M5/org.reactfx.util.SparseList.getPresent(SparseList.java:259)
	at reactfx@2.0-M5/org.reactfx.collection.MemoizationListImpl$MemoizedView.get(MemoizationList.java:41)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledArea.visibleParToAllParIndex(GenericStyledArea.java:990)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.ViewActions.lastVisibleParToAllParIndex(ViewActions.java:289)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledArea.nextPage(GenericStyledArea.java:1422)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledAreaBehavior.lambda$static$10(GenericStyledAreaBehavior.java:107)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate.lambda$consume$0(InputMapTemplate.java:221)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.PatternActionTemplate.lambda$null$1(InputMapTemplate.java:425)
	at java.base/java.util.Optional.map(Optional.java:260)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.PatternActionTemplate.lambda$getInputHandlerTemplateMap$2(InputMapTemplate.java:425)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:25)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:25)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate$1.lambda$null$0(InputMapTemplate.java:202)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate$HandlerTemplateConsumer$1.lambda$accept$0(InputMapTemplate.java:103)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.InputHandler.handle(InputHandler.java:50)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:247)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
	at javafx.base@19-ea/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
	at javafx.base@19-ea/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base@19-ea/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.base@19-ea/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.graphics@19-ea/javafx.scene.Scene$KeyHandler.process(Scene.java:4089)
	at javafx.graphics@19-ea/javafx.scene.Scene.processKeyEvent(Scene.java:2146)
	at javafx.graphics@19-ea/javafx.scene.Scene$ScenePeerListener.keyEvent(Scene.java:2614)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:218)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:150)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleKeyEvent$1(GlassViewEventHandler.java:250)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:424)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleKeyEvent(GlassViewEventHandler.java:249)
	at javafx.graphics@19-ea/com.sun.glass.ui.View.handleKeyEvent(View.java:542)
	at javafx.graphics@19-ea/com.sun.glass.ui.View.notifyKey(View.java:966)
	at javafx.graphics@19-ea/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at javafx.graphics@19-ea/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:316)
	at java.base/java.lang.Thread.run(Thread.java:1583)

2023-11-12 15:12:41.862 [ERROR] [JavaFX Application Thread] com.foo.FxPlatform - Error in FxApplicationjava.util.NoSuchElementException
	at reactfx@2.0-M5/org.reactfx.util.SparseList$AbsentSegment.getOrThrow(SparseList.java:78)
	at reactfx@2.0-M5/org.reactfx.util.FingerTree.lambda$get$2(FingerTree.java:1027)
	at reactfx@2.0-M5/org.reactfx.util.BiIndex.map(BiIndex.java:15)
	at reactfx@2.0-M5/org.reactfx.util.FingerTree.get(FingerTree.java:1027)
	at reactfx@2.0-M5/org.reactfx.util.SparseList.getPresent(SparseList.java:259)
	at reactfx@2.0-M5/org.reactfx.collection.MemoizationListImpl$MemoizedView.get(MemoizationList.java:41)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledArea.visibleParToAllParIndex(GenericStyledArea.java:990)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.ViewActions.lastVisibleParToAllParIndex(ViewActions.java:289)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledArea.nextPage(GenericStyledArea.java:1422)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledAreaBehavior.lambda$static$10(GenericStyledAreaBehavior.java:107)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate.lambda$consume$0(InputMapTemplate.java:221)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.PatternActionTemplate.lambda$null$1(InputMapTemplate.java:425)
	at java.base/java.util.Optional.map(Optional.java:260)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.PatternActionTemplate.lambda$getInputHandlerTemplateMap$2(InputMapTemplate.java:425)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:25)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:25)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate$1.lambda$null$0(InputMapTemplate.java:202)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate$HandlerTemplateConsumer$1.lambda$accept$0(InputMapTemplate.java:103)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.InputHandler.handle(InputHandler.java:50)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:247)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
	at javafx.base@19-ea/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
	at javafx.base@19-ea/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base@19-ea/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.base@19-ea/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.graphics@19-ea/javafx.scene.Scene$KeyHandler.process(Scene.java:4089)
	at javafx.graphics@19-ea/javafx.scene.Scene.processKeyEvent(Scene.java:2146)
	at javafx.graphics@19-ea/javafx.scene.Scene$ScenePeerListener.keyEvent(Scene.java:2614)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:218)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:150)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleKeyEvent$1(GlassViewEventHandler.java:250)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:424)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleKeyEvent(GlassViewEventHandler.java:249)
	at javafx.graphics@19-ea/com.sun.glass.ui.View.handleKeyEvent(View.java:542)
	at javafx.graphics@19-ea/com.sun.glass.ui.View.notifyKey(View.java:966)
	at javafx.graphics@19-ea/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at javafx.graphics@19-ea/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:316)
	at java.base/java.lang.Thread.run(Thread.java:1583)

2023-11-12 15:12:43.571 [ERROR] [JavaFX Application Thread] com.foo.FxPlatform - Error in FxApplicationjava.lang.IndexOutOfBoundsException: 39 not in [0, 38]
	at reactfx@2.0-M5/org.reactfx.util.Lists.checkPosition(Lists.java:110)
	at reactfx@2.0-M5/org.reactfx.util.Lists.checkPosition(Lists.java:105)
	at reactfx@2.0-M5/org.reactfx.util.FingerTree$NonEmptyFingerTree.locateProgressively(FingerTree.java:51)
	at reactfx@2.0-M5/org.reactfx.util.FingerTree.get(FingerTree.java:1026)
	at reactfx@2.0-M5/org.reactfx.util.SparseList.getPresent(SparseList.java:259)
	at reactfx@2.0-M5/org.reactfx.collection.MemoizationListImpl$MemoizedView.get(MemoizationList.java:41)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledArea.visibleParToAllParIndex(GenericStyledArea.java:990)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.ViewActions.lastVisibleParToAllParIndex(ViewActions.java:289)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledArea.nextPage(GenericStyledArea.java:1422)
	at org.fxmisc.richtext@0.10.9/org.fxmisc.richtext.GenericStyledAreaBehavior.lambda$static$10(GenericStyledAreaBehavior.java:107)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate.lambda$consume$0(InputMapTemplate.java:221)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.PatternActionTemplate.lambda$null$1(InputMapTemplate.java:425)
	at java.base/java.util.Optional.map(Optional.java:260)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.PatternActionTemplate.lambda$getInputHandlerTemplateMap$2(InputMapTemplate.java:425)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:25)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:25)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate$1.lambda$null$0(InputMapTemplate.java:202)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputHandlerTemplateMap.lambda$sequence$0(InputHandlerTemplateMap.java:24)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.template.InputMapTemplate$HandlerTemplateConsumer$1.lambda$accept$0(InputMapTemplate.java:103)
	at wellbehavedfx@0.3.3/org.fxmisc.wellbehaved.event.InputHandler.handle(InputHandler.java:50)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:247)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
	at javafx.base@19-ea/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
	at javafx.base@19-ea/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at javafx.base@19-ea/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at javafx.base@19-ea/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at javafx.base@19-ea/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at javafx.base@19-ea/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.base@19-ea/javafx.event.Event.fireEvent(Event.java:198)
	at javafx.graphics@19-ea/javafx.scene.Scene$KeyHandler.process(Scene.java:4089)
	at javafx.graphics@19-ea/javafx.scene.Scene.processKeyEvent(Scene.java:2146)
	at javafx.graphics@19-ea/javafx.scene.Scene$ScenePeerListener.keyEvent(Scene.java:2614)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:218)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:150)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleKeyEvent$1(GlassViewEventHandler.java:250)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:424)
	at javafx.graphics@19-ea/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleKeyEvent(GlassViewEventHandler.java:249)
	at javafx.graphics@19-ea/com.sun.glass.ui.View.handleKeyEvent(View.java:542)
	at javafx.graphics@19-ea/com.sun.glass.ui.View.notifyKey(View.java:966)
	at javafx.graphics@19-ea/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at javafx.graphics@19-ea/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:316)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

--- a/richtextfx/src/main/java/org/fxmisc/richtext/GenericStyledArea.java
+++ b/richtextfx/src/main/java/org/fxmisc/richtext/GenericStyledArea.java
@@ -8,6 +8,7 @@ import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 import java.util.Locale;
+import java.util.NoSuchElementException;
 import java.util.Objects;
 import java.util.Optional;
 import java.util.function.BiConsumer;
@@ -985,13 +986,19 @@ public class GenericStyledArea<PS, SEG, S> extends Region
             );
         }
 
-        Cell<Paragraph<PS,SEG,S>, ParagraphBox<PS,SEG,S>> visibleCell = null;
+        try {
+            Cell<Paragraph<PS,SEG,S>, ParagraphBox<PS,SEG,S>> visibleCell = null;
 
-        if ( visibleParIndex > 0 ) visibleCell = virtualFlow.visibleCells().get( visibleParIndex );
-        else visibleCell = virtualFlow.getCellIfVisible( virtualFlow.getFirstVisibleIndex() )
-            .orElseGet( () -> virtualFlow.visibleCells().get( visibleParIndex ) );
+            if ( visibleParIndex > 0 ) visibleCell = virtualFlow.visibleCells().get( visibleParIndex );
+            else visibleCell = virtualFlow.getCellIfVisible( virtualFlow.getFirstVisibleIndex() )
+                       .orElseGet( () -> virtualFlow.visibleCells().get( visibleParIndex ) );
 
         return visibleCell.getNode().getIndex();
+        }
+        catch ( NoSuchElementException EX )
+        {
+            return -1;
+        }
     }
 
     @Override

--- a/richtextfx/src/main/java/org/fxmisc/richtext/GenericStyledArea.java
+++ b/richtextfx/src/main/java/org/fxmisc/richtext/GenericStyledArea.java
@@ -995,7 +995,7 @@ public class GenericStyledArea<PS, SEG, S> extends Region
 
         return visibleCell.getNode().getIndex();
         }
-        catch ( NoSuchElementException EX )
+        catch ( IndexOutOfBoundsException | NoSuchElementException EX )
         {
             return -1;
         }

